libs/langchain/langchain/chat_models/fireworks.py

```diff
@@ -111,6 +124,197 @@ def _llm_type(self) -> str:
         """Return type of llm."""
         return "fireworks-chat"
 
+    def generate(
+        self,
+        messages: List[List[BaseMessage]],
+        stop: Optional[List[str]] = None,
+        callbacks: Callbacks = None,
+        *,
+        tags: Optional[List[str]] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+        run_name: Optional[str] = None,
+        **kwargs: Any,
+    ) -> LLMResult:
+        """Top Level call"""
+        params = self._get_invocation_params(stop=stop, **kwargs)
+        options = {"stop": stop}
+
+        callback_manager = CallbackManager.configure(
+            callbacks,
+            self.callbacks,
+            self.verbose,
+            tags,
+            self.tags,
+            metadata,
+            self.metadata,
+        )
+        run_managers = callback_manager.on_chat_model_start(
+            dumpd(self),
+            messages,
+            invocation_params=params,
+            options=options,
+            name=run_name,
+        )
+
+        def _completion_with_retry_batching(
+            message: List[List[BaseMessage]],
+        ) -> List[Any]:
+            args_list = [
+                (m, stop, run_managers[i] if run_managers else None, kwargs)
+                for i, m in enumerate(message)
+            ]
+            with ThreadPoolExecutor() as executor:
+                results = list(executor.map(self._process_message, args_list))
+
+            return results
+
+        sub_messages = self.get_batch_messages(params, messages, stop)
+
+        results = []
+        for message in sub_messages:
+            results.extend(_completion_with_retry_batching(message))
+
+        flattened_outputs = [
+            LLMResult(generations=[res.generations], llm_output=res.llm_output)
+            for res in results
+        ]
+        llm_output = self._combine_llm_outputs([res.llm_output for res in results])
+        generations = [res.generations for res in results]
+        output = LLMResult(generations=generations, llm_output=llm_output)
+        if run_managers:
+            run_infos = []
+            for manager, flattened_output in zip(run_managers, flattened_outputs):
+                manager.on_llm_end(flattened_output)
+                run_infos.append(RunInfo(run_id=manager.run_id))
+            output.run = run_infos
+        return output
+
+    async def agenerate(
```

Comment:
The base class has a default implementation of `agenerate` and `generate`. This class shuold not need to re-implement. 

What's the problem with the existing implementation?

Reply:
We want a batching version of `generate()` as the default implementation process serially. Does it make sense to implement in this way?

Reply:
Why do you want the default implementation to process serially rather than in parallel?

Reply:
The reason should be the same as the reason for llms batching. We want faster completions when we have multiple messages input. cc @tjaffri

Reply:
Default `generate` implementation doesn't call `batch`. Thus, it is still serial process.

Reply:
is it possible to call the `batch` method directly? or in other words, why are you calling `generate` directly?

Reply:
I will drop `generate/agenerate`  and use `.batch` for batching instead.